src/Command/ExportCommand.php

```diff
@@ -19,15 +21,23 @@ class ExportCommand extends Command {
 	/** @var GeneratorSelector */
 	private $generatorSelector;
 
+	/** @var ReplicasClient */
+	private $replicasClients;
+
+	/** @var CreditRepository */
+	private $creditRepo;
+
 	/** @var Api */
 	private $api;
 
 	/** @var bool */
 	private $enableCache;
 
-	public function __construct( GeneratorSelector $generatorSelector, Api $api, bool $enableCache ) {
+	public function __construct( GeneratorSelector $generatorSelector, ReplicasClient $replicasClient, CreditRepository $creditRepo, Api $api, bool $enableCache ) {
```

Comment:
Not critical, but another option may be to only inject `ReplicasClient` in the `CreditRepository` constructor, since that's the only place it's used (if I'm reading the code correctly). That way you don't need to pass around the `ReplicasClient` instance.

So CreditRepository could have:
```
/** @var ReplicasClient $client */
private $client;

public function __construct( ReplicasClient $client ) {
	$this->client = $client;
}
```
That may or may not work out-of-the-box, but if not, you can specify this dependency in services.yaml like:

```
services:
	...
	App\Repository\CreditRepository:
		arguments:
			- '@Wikimedia\ToolforgeBundle\Service\ReplicasClient'
```

Reply:
I had tried your first suggestion, and for whatever reason I couldn't get it to work. I tried it again and it worked! :)